engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/StringLiteralBranchNode.java

```diff
@@ -39,9 +42,26 @@ void doText(
     }
   }
 
+  @Specialization(
+      guards = {"targetInterop.isString(target)"},
+      limit = "3")
+  void doInteropString(
+      VirtualFrame frame,
+      Object state,
+      Object target,
```

Comment:
With [my changes](https://github.com/enso-org/enso/pull/9203#issuecomment-1968896105) this specialization isn't needed. It may be needed in other cases, but we don't have a test for it yet. If a `TruffleObject` from other language which `isString` is passed in, then such a specialization might get triggered.

I'd probably change the `target` type to `TruffleObject` and added a guard `!target instanceof EnsoObject`... but it is just a speculation without having a real test.

Reply:
I was mostly inspired by `EqualsSimpleNode`.

There we **do have** a `equalsStrings` that is relying on `selfInterop.isString`.

I think that case-of should be consistent with Equals in that regard, even if we don't have a test showing it.

Alternatively, if the `coerceNode` is supposed to be enough of a fix, maybe that means that we could remove this specialization from `EqualsSimpleNode`? What do you think?

Reply:
Is that OK?

Reply:
It's okeyish.